src/visibility.js

```diff
@@ -140,14 +140,20 @@ class Visibility extends BasePlugin {
   }
 
   _stopFloating() {
-    Utils.Dom.removeClassName(this._floatingPoster, FLOATING_POSTER_CLASS_SHOW);
-    Utils.Dom.removeClassName(this._floatingContainer, FLOATING_ACTIVE_CLASS);
-    Utils.Dom.removeAttribute(this._floatingContainer, 'style');
-    if (this.config.draggable) {
-      this.eventManager.unlisten(this._floatingContainer, 'mousedown');
-      this.eventManager.unlisten(this._floatingContainer, 'touchstart');
-      this._stopDrag();
-    }
+    const floatingPosterElements = document.querySelectorAll(`.${FLOATING_POSTER_CLASS}`);
```

Comment:
why does the x cause the previous floating player to be closed and not the current one ?

Reply:
cause it built only one time and not for every floating player.

Reply:
the x button is built one time ? can you link where it is built ?

Reply:
ok so after check this again, it was not created only once but in every click on x button the values in this._floatingPoster and this._floatingContainer (here - https://github.com/kaltura/playkit-js-visibility/blob/master/src/visibility.js#L143) contains the floating player from the first entry

Reply:
2 questions:
1. do we want to just hide any previous floating player when we show a new floating player?
2. can we just do this by going over all the players on the page (KalturaPlayer.getPlayers) and calling enterPictureInPicture and then immediately exitPictureInPicture?
3. can we just pause the playback of the other players and then it will hide their floating player ?